packages/flutter/lib/src/widgets/stretch_effect.dart

```diff
@@ -0,0 +1,251 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'image_filter.dart';
+
+/// A widget that applies a stretching visual effect to its child.
+///
+/// When shader-based effects are supported, they are used for better performance
+/// and visual quality. Otherwise, a matrix transform is applied as a fallback.
+///
+/// Used by [StretchingOverscrollIndicator] widget.
+class StretchEffect extends StatelessWidget {
+  /// Creates a [StretchEffect] widget that applies a stretch effect
+  /// when the user overscrolls horizontally or vertically.
+  ///
+  /// The [stretchStrength] controls the intensity of the stretch effect
+  /// and must be between -1.0 and 1.0.
+  const StretchEffect({
+    super.key,
+    this.stretchStrength = 0.0,
+    required this.axis,
+    required this.child,
+  }) : assert(
+         stretchStrength >= -1.0 && stretchStrength <= 1.0,
+         'stretchStrength must be between -1.0 and 1.0',
+       );
+
+  /// The overscroll strength applied for the stretching effect.
+  ///
+  /// The value should be between -1.0 and 1.0 inclusive.
+  /// For horizontal axis, Positive values apply a pull from
+  /// left to right, while negative values pull from right to left.
```

Comment:
Can you describe what 1.0 stands for, so that developers that use this widget independently know how to derive this value?

Also for parity, you should probably also explain vertical here.